src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java

```diff
@@ -416,4 +440,30 @@ private static EndpointDescriptor makeDestination(FlowEndpoint endpoint) {
             return new EndpointDescriptor(endpoint, "destination");
         }
     }
+
+    /**
+     * Validate the flow mirror point.
+     */
+    public void flowMirrorPointValidate(RequestedFlowMirrorPoint mirrorPoint)
+            throws InvalidFlowException, UnavailableFlowEndpointException {
+
+        checkSwitchesExistsAndActive(mirrorPoint.getMirrorPointSwitchId(), mirrorPoint.getSinkEndpointSwitchId());
+
+        EndpointDescriptor descriptor = EndpointDescriptor.makeDestination(FlowEndpoint.builder()
+                .switchId(mirrorPoint.getSinkEndpointSwitchId())
+                .portNumber(mirrorPoint.getSinkEndpointPort())
+                .outerVlanId(mirrorPoint.getSinkEndpointVlan())
+                .build());
+
+        SwitchId switchId = descriptor.endpoint.getSwitchId();
+        SwitchProperties properties = switchPropertiesRepository.findBySwitchId(switchId)
+                .orElseThrow(() -> new InvalidFlowException(
+                        format("Couldn't get switch properties for %s switch %s.", descriptor.name, switchId),
+                        ErrorType.DATA_INVALID));
+
+        checkForMultiTableRequirement(descriptor, properties);
+        checkFlowForIslConflicts(descriptor);
+        checkFlowForFlowConflicts(mirrorPoint.getMirrorPointId(), descriptor);
+        checkFlowForSinkEndpointConflicts(mirrorPoint.getMirrorPointId(), descriptor);
```

Comment:
What about check for conflict between flow endpoint and mirror sink endpoint on flow manipulations?

Reply:
This check has been added.